hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -1179,4 +1179,25 @@ public void testDatabaseLocationWithSlashInWarehouseDir() {
 
     assertThat(database.getLocationUri()).isEqualTo("s3://bucket/database.db");
   }
+
+  @Test
+  public void testListNamespacesWithEmptyNamespace() {
+    Namespace namespaceWithOneLevel = Namespace.of("parent");
+    assertThat(catalog.listNamespaces(namespaceWithOneLevel))
+        .as("Namespace with one level will return zero namespace.")
+        .hasSize(0);
+
+    Namespace namespaceWithTwoLevel = Namespace.of("parent", "child1");
+    assertThatThrownBy(() -> catalog.listNamespaces(namespaceWithTwoLevel))
+        .isInstanceOf(NoSuchNamespaceException.class)
+        .hasMessageContaining("Namespace does not exist");
+
+    assertThat(catalog.listNamespaces(Namespace.empty()))
+        .as("Empty namespace should return two namespaces.")
+        .hasSize(2);
```

Comment:
which namespaces are being returned here? I think the check should include what's being returned: `.containsExactly(...)`